rma_sale/tests/test_rma_sale.py

```diff
@@ -91,11 +92,18 @@ def test_create_rma_from_so(self):
             {"login": "test_refund_with_so", "name": "Test"}
         )
         order.user_id = user.id
+        order.analytic_account_id = self.env.ref(
+            "analytic.analytic_our_super_product"
+        ).id
         rma.action_confirm()
         rma.reception_move_id.quantity_done = rma.product_uom_qty
         rma.reception_move_id.picking_id._action_done()
         rma.action_refund()
         self.assertEqual(rma.refund_id.user_id, user)
+        self.assertEqual(
+            rma.refund_id.invoice_line_ids.analytic_account_id,
```

Comment:
For tests, use always hard-coded references. Here, you are reimplementing again the same logic of what you want to test.

Reply:
What would be the difference in doing it like
```
analytic_account = some_account
order.analytic_account_id = analytic_account.id
rma.action_refund()
self.assertEqual(rma.refund_id.invoice_line_ids.analytic_account_id, analytic_account)
```
It is basically the same .

Reply:
Both makes the test green right now, but it's not a guideline of good practice.

1. Readability. In its current form, it requires to "decode" the logic behind the data that you want to check, requiring to read upper lines until finding the line where the value is assigned.
2. Scope. You don't have to test the implementation, but the business requirement, which is to have the analytic account in the refund line, so this is what you have to check.
3. Avoid side effects. If the implementation changes, you are still testing the final result, which is the business requirement.
4. Resiliency. If tomorrow the field name changes, you won't need to touch in 2 places, only in the first analytic assignation.

Do you understand now why it's a good practice?

Reply:
In this test case i think it is more readable, because what i want to test is.
If the refund line analytic account is the same as the order analytic account.
```
self.assertEqual(
    rma.refund_id.invoice_line_ids.analytic_account_id,
    order.analytic_account_id,
)
```
With this line
```
self.assertEqual(
    rma.refund_id.invoice_line_ids.analytic_account_id,
    analytic_account,
)
```
i have to look, where analytic_account is defined. 

You are right if the field would change, i have to change it in this test. Two times, but for me this would be fine.

Reply:
Most IDEs have the option to go to the definition, so you can easily navigate to `analytic_account` assignation, so that's not a big problem. On contrary, using a complex expression like `order.analytic_account_id` is not traceable and may vary if other code reassigns this value. Imagine a test failure that gets you crazy because there's an installed module that is auto-assigning the analytic account in the sales order according some criteria after you have made the refund.

As said, these are good practices for avoiding risks. That risks may not come, but if so, it can make you lose a lot of time debugging and trying to figure out things.

Reply:
I don't think `order.analytic_account_id` is complex. 
Also i don't think we should consider other modules which will maybe change the behavior als long as the not a dependency or a auto_install module of the tested one.
Also it could be the other way around, if some module changes the account on the sale.order within the action_refund before the main code is executed. Then it its fine that initial account which was set in the test is not the one we test against.

Reply:
Well, I desist trying to explain why it's convenient. I'm not going to block the PR due to this.